src/components/ContentOpenWith/ContentOpenWith.js

```diff
@@ -229,21 +238,24 @@ class ContentOpenWith extends PureComponent<Props, State> {
                 this.executeIntegrationSuccessHandler,
                 this.executeIntegrationErrorHandler,
             );
+
+        this.executeId = appIntegrationId;
     };
 
     /**
      * Opens the integration in a new tab based on the API data
      *
      * @private
-     * @param {ExecuteAPI} executePostData - API response on how to open an executed integration
+     * @param {ExecuteAPI} executeData - API response on how to open an executed integration
 
      * @return {void}
      */
-    executeIntegrationSuccessHandler = (executePostData: ExecuteAPI): void => {
-        const { method, url } = executePostData;
+    executeIntegrationSuccessHandler = (executeData: ExecuteAPI): void => {
+        const { onExecute }: Props = this.props;
+        const { method, url } = executeData;
         switch (method) {
             case HTTP_POST:
-                this.setState({ executePostData });
+                this.setState({ executePostData: executeData });
```

Comment:
Should the state property also be renamed?

Reply:
the state property is only in the post case, while as this data from the API could be used for POST or GET. Hence I've left it general here but made it more specific in state